### ISSUE TITLE
[Bug] Disabling entry filters isn't possible when allowOverrideEntryFilters=true and entryFilterNames is specified in broker.conf

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#20520]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

3.0.x, master

### Minimal reproduce step

Disabling entry filters isn't possible when `allowOverrideEntryFilters=true` and `entryFilterNames` is specified in `broker.conf`.

### What did you expect to see?

It should be possible to disable entry filters completely at topic level when `allowOverrideEntryFilters=true`

### What did you see instead?

The `entryFilterNames` in `broker.conf` are used when the list of entry filters for the topic is empty. Please see the code:
https://github.com/apache/pulsar/blob/50b9a93e42e412d9f17b1637287d1a4c7c7ab148/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/EntryFilterSupport.java#L45-L51

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by eolivelli** [comment]
I asked for this feature,  because in a shared environment the amministrator wants to enforce some filters without allowing users to disable them. 

I think that the current behavior is correct

**Comment by lhotari** [comment]
> I asked for this feature, because in a shared environment the amministrator wants to enforce some filters without allowing users to disable them.
> 
> I think that the current behavior is correct

@eolivelli Yes. However, when `allowOverrideEntryFilters=true`, shouldn't it be possible for users to disable them?
